From 0e366d11fe4e7d53543de2e52fd9f44f908effea Mon Sep 17 00:00:00 2001
From: Huibin Hong <huibin.hong@rock-chips.com>
Date: Wed, 12 Jul 2017 10:22:18 +0800
Subject: [PATCH] spi: rockchip: rockchip_spi_prepare_dma return 1 if it is ok

If rockchip_spi_prepare_dma return 0, spi_transfer_one_message
won't wait for the dma complete event. Which may cause unknown
error.

Change-Id: Idf51a737bbd9fa9b9d1ae749636fe524de1cbfa9
Signed-off-by: Huibin Hong <huibin.hong@rock-chips.com>
---
 drivers/spi/spi-rockchip.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi-rockchip.c b/drivers/spi/spi-rockchip.c
index 42c8fcd82758..573b8a4b4cc9 100644
--- a/drivers/spi/spi-rockchip.c
+++ b/drivers/spi/spi-rockchip.c
@@ -499,7 +499,7 @@ static int rockchip_spi_prepare_dma(struct rockchip_spi *rs)
 		dma_async_issue_pending(rs->dma_tx.ch);
 	}
 
-	return 0;
+	return 1;
 }
 
 static void rockchip_spi_config(struct rockchip_spi *rs)
-- 
2.35.3

